timeseries/src/autogluon/timeseries/trainer.py

```diff
@@ -1313,21 +1221,69 @@ def refit_full(self, model: str = "all") -> Dict[str, str]:
         logger.info(f"Total runtime: {time.time() - time_start:.2f} s")
         return copy.deepcopy(self.model_refit_map)
 
-    def construct_model_templates(
-        self, hyperparameters: Union[str, Dict[str, Any]], multi_window: bool = False, **kwargs
-    ) -> List[AbstractTimeSeriesModel]:
-        """Constructs a list of unfit models based on the hyperparameters dict."""
-        raise NotImplementedError
+    def construct_model_templates(self, hyperparameters: Union[str, Dict], multi_window: bool = False, **kwargs):
+        path = kwargs.pop("path", self.path)
+        eval_metric = kwargs.pop("eval_metric", self.eval_metric)
+        eval_metric_seasonal_period = kwargs.pop("eval_metric", self.eval_metric_seasonal_period)
+        quantile_levels = kwargs.pop("quantile_levels", self.quantile_levels)
+        hyperparameter_tune = kwargs.get("hyperparameter_tune", False)
+
+        return get_preset_models(
+            path=path,
+            eval_metric=eval_metric,
+            eval_metric_seasonal_period=eval_metric_seasonal_period,
+            prediction_length=self.prediction_length,
+            freq=kwargs.get("freq"),
+            hyperparameters=hyperparameters,
+            hyperparameter_tune=hyperparameter_tune,
+            quantile_levels=quantile_levels,
+            all_assigned_names=self._get_banned_model_names(),
+            target=self.target,
+            metadata=self.metadata,
+            excluded_model_types=kwargs.get("excluded_model_types"),
+            # if skip_model_selection = True, we skip backtesting
+            multi_window=multi_window and not self.skip_model_selection,
+        )
 
     def fit(
         self,
         train_data: TimeSeriesDataFrame,
-        hyperparameters: Union[str, Dict[str, Any]],
+        hyperparameters: Union[str, Dict[Any, Dict]],
         val_data: Optional[TimeSeriesDataFrame] = None,
-        **kwargs,
-    ) -> None:
-        raise NotImplementedError
-
-    # TODO: def _filter_base_models_via_infer_limit
+        hyperparameter_tune_kwargs: Optional[Union[str, Dict]] = None,
+        excluded_model_types: Optional[List[str]] = None,
+        time_limit: Optional[float] = None,
+        random_seed: Optional[int] = None,
+    ):
+        """
+        Fit a set of timeseries models specified by the `hyperparameters`
+        dictionary that maps model names to their specified hyperparameters.
 
-    # TODO: persist and unpersist models
+        Parameters
+        ----------
+        train_data: TimeSeriesDataFrame
```

Comment:
Nit: Let's remove the type hints from the docstrings, they are already available in the method signature?